doc/source/ray-air/check-ingest.rst

```diff
@@ -29,175 +22,84 @@ user-defined function to preprocess batches of data, and (3) runs an AIR Trainer
 
 .. tabbed:: Bulk Ingest
 
-    This example shows bulk ingest (the default). Data is bulk loaded and made available
-    directly via a ``Dataset`` object that can be looped over manually.
+    By default, AIR loads all datasets into the Ray object store at the start of training.
+    This provides the best performance if the cluster can fit the datasets
+    entirely in memory, or if the preprocessing step is expensive to run more than once.
 
     .. literalinclude:: doc_code/air_ingest.py
         :language: python
         :start-after: __config_4__
         :end-before: __config_4_end__
 
-.. tabbed:: Streaming Ingest
+    You should use bulk ingest when:
+
+     * you have enough memory to fit data blocks in cluster object store;
+     * your preprocessing step is expensive per each epoch; and
+     * you want best performance when both or either the above conditions are met.
+
+.. tabbed:: Streaming Ingest (experimental)
+
+    In streaming ingest mode, ``get_dataset_shard`` returns a ``DatasetPipeline`` pipeline that
+    can be used to read data in a streaming way.
+    To enable streaming ingest, set ``use_stream_api=True`` in the dataset config.
 
-    This example shows enabling streaming ingest for the "train" dataset with a *N-byte* window.
-    This means that AIR will only load *N* bytes of data from the datasource at a time (the data
-    may be larger once deserialized in memory or if individual files are larger than the window).
+    By default, this will tell AIR to load *windows* of 1GiB of data into memory at a time.
+    Performance can be increased with larger window sizes, which can be adjusted using the
+    ``stream_window_size`` config.
+    A reasonable stream window size is something like 20% of available object store memory.
 
     .. literalinclude:: doc_code/air_ingest.py
         :language: python
         :start-after: __config_5__
         :end-before: __config_5_end__
 
+    Use streaming ingest when:
+
+     * you have large datasets that don't fit into memory;
+     * you want to process small chunks or blocks per window;
+     * you can use small windows with small data blocks minimizing or avoiding memory starvation or OOM errors; and
+     * your preprocessing step is not a bottleneck or not an expensive operation since it's re-executed on each pass over the data.
+
 Shuffling Data
 ~~~~~~~~~~~~~~
 
-AIR offers several options for per-epoch shuffling, including *local
-(per-shard) shuffling*  and *global (whole-dataset) shuffling*.
+Shuffling or data randomization is important for training high-quality models. By default, AIR will randomize the order the data files (blocks) are read from. AIR also offers options for further randomizing data records within each file:
 
 .. tabbed:: Local Shuffling
+    
+    Local shuffling is the recommended approach for randomizing data order. To use local shuffle,
+    simply specify a non-zero ``local_shuffle_buffer_size`` as an argument to ``iter_batches()``.
```

Comment:
nit: this must actually be larger than `batch_size`, not just non-zero.

Reply:
I think that's allowed right? It's just internally rounded up to batch size.

Reply:
Filed https://github.com/ray-project/ray/pull/26915/files